server/sonar-server/src/test/java/org/sonar/server/computation/task/projectanalysis/webhook/WebhookPayloadTest.java

```diff
@@ -0,0 +1,137 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.computation.task.projectanalysis.webhook;
+
+import java.util.Date;
+import java.util.Optional;
+import org.junit.Test;
+import org.sonar.api.ce.posttask.CeTask;
+import org.sonar.api.ce.posttask.PostProjectAnalysisTask;
+import org.sonar.api.ce.posttask.Project;
+import org.sonar.api.ce.posttask.QualityGate;
+import org.sonar.api.ce.posttask.ScannerContext;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.sonar.api.ce.posttask.PostProjectAnalysisTaskTester.newCeTaskBuilder;
+import static org.sonar.api.ce.posttask.PostProjectAnalysisTaskTester.newConditionBuilder;
+import static org.sonar.api.ce.posttask.PostProjectAnalysisTaskTester.newProjectBuilder;
+import static org.sonar.api.ce.posttask.PostProjectAnalysisTaskTester.newQualityGateBuilder;
+import static org.sonar.test.JsonAssert.assertJson;
+
+public class WebhookPayloadTest {
+
+  @Test
+  public void create_payload_for_successful_analysis() {
+    PostProjectAnalysisTask.ProjectAnalysis analysis = new PostProjectAnalysisTask.ProjectAnalysis() {
+      @Override
+      public CeTask getCeTask() {
+        return newCeTaskBuilder()
+          .setStatus(CeTask.Status.SUCCESS)
+          .setId("#1")
+          .build();
+      }
+
+      @Override
+      public Project getProject() {
+        return newProjectBuilder()
+          .setUuid("P1_UUID")
+          .setKey("P1")
+          .setName("Project One")
+          .build();
+      }
+
+      @Override
+      public QualityGate getQualityGate() {
+        return newQualityGateBuilder()
+          .setId("G1")
+          .setName("Gate One")
+          .setStatus(QualityGate.Status.WARN)
+          .add(newConditionBuilder()
+            .setMetricKey("coverage")
+            .setOperator(QualityGate.Operator.GREATER_THAN)
+            .setOnLeakPeriod(true)
+            .setWarningThreshold("75.0")
+            .setErrorThreshold("70.0")
+            .build(QualityGate.EvaluationStatus.WARN, "74.0"))
+          .build();
+      }
+
+      @Override
+      public Date getDate() {
+        throw new UnsupportedOperationException();
+      }
+
+      @Override
+      public Optional<Date> getAnalysisDate() {
+        return Optional.of(new Date(1_500_000_000_000L));
+      }
+
+      @Override
+      public ScannerContext getScannerContext() {
+        throw new UnsupportedOperationException();
+      }
+    };
+    WebhookPayload payload = WebhookPayload.from(analysis);
+    assertThat(payload.getProjectKey()).isEqualTo("P1");
+    assertJson(payload.toJson()).isSimilarTo(getClass().getResource("WebhookPayloadTest/success.json"));
```

Comment:
I tried and I can say that no. It does not help at all as JSON double-quotes must be escaped.